ci/scripts/utils/publish_logs.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+
+import os
+import secrets
+import re
+import string
+from githubpr import GitHubPR, GitHubDBError
+from argparse import ArgumentParser, ArgumentTypeError, FileType
+
+
+def is_file(path):
+    """
+    Check if path is an existing file.
+
+    Parameters
+    ----------
+    path : str
+        The path to check.
+
+    Returns
+    -------
+    str
+        The path, if it is an existing file.
+
+    Raises
+    ------
+    argparse.ArgumentTypeError
+        If the path is not an existing file.
+    """
+
+    if not os.path.isfile(path):
+        raise ArgumentTypeError(f"{path} does not exist or is not a file.")
+    return path
+
+
+def input_args():
+    """
+    Parse command line arguments.
+
+    Returns
+    -------
+    argparse.Namespace
+        The parsed command line arguments.
+    """
+
+    description = """Arguments for creating and updating error log files
+    """
+    parser = ArgumentParser(description=description)
+
+    parser.add_argument('--file', help='path to file for uploading to GitHub', required=False, type=FileType('r'), nargs='+')
+    parser.add_argument('--gist', help='create a gist of the file', nargs=1, metavar='identifier_string', required=False)
+    parser.add_argument('--repo', help='create a file in a repo', nargs=1, metavar='path_header', required=False)
+    parser.add_argument('--delete_gists', help='deletes all the gits from authenticated user', action='store_true', default=False, required=False)
```

Comment:
Are these mutually exclusive? Should this be modal like setup_exp?

Reply:
They are not.  You can use and/or `--gist` and `--repo`.